### PR TITLE
Fix: Assets page for multisig

### LINKED ISSUE #2656: Multisig. Old multisig wallets become unusable
## Description

Portfolio view contains no chains for single Multisig wallets, which was removed from config

https://github.com/user-attachments/assets/97c545b2-3674-47ac-9b0c-270c8f76f253

## Scenario:
1. Add multisig to any chain
2. Remove multisig flag for this chain in config
3. Restart the app

## Expected results:
Old multisigs should work

## Actual result:
Portfolio view is empty

## Environment
https://github.com/novasamatech/nova-spektr/commit/993f059a3cee32a9324eb293a40652e4ca3b07cc

### PR DESCRIPTION
**Pull request by tuul-wq**
- Show chain for Multisig wallet with a single chain (that had msig option)

closes #2656

### CODE REVIEW

File: src/renderer/features/assets/AssetsChainView/ui/AssetsChainView.tsx [hunk 1]
```diff
@@ -34,26 +34,23 @@ export const AssetsChainView = ({ query, activeShards, hideZeroBalances, assetsV
...
         );
       });
     });
 
     const filteredChains = availableChains.filter((c) => {
-      if (!connections[c.chainId]) {
-        return false;
-      }
+      if (!connections[c.chainId]) return false;
+      if (networkUtils.isDisabledConnection(connections[c.chainId])) return false;
```
Comment (johnthecat) [inline src/renderer/features/assets/AssetsChainView/ui/AssetsChainView.tsx]: Check that connection exist before checking status

File: src/renderer/features/assets/AssetsPortfolioView/model/portfolio-model.ts [hunk 2]
```diff
@@ -105,22 +108,25 @@ sample({
   },
   fn: ({ connections, chains, tokens, activeWallet }): AssetByChains[] => {
     const isMultisigWallet = walletUtils.isMultisig(activeWallet);
+    const multisigChainToInclude = isMultisigWallet ? activeWallet.accounts[0].chainId : undefined;
```
Comment (johnthecat) [inline src/renderer/features/assets/AssetsPortfolioView/model/portfolio-model.ts]: Plz check account before diving into chainId

File: src/renderer/features/assets/AssetsPortfolioView/model/portfolio-model.ts [hunk 3]
```diff
@@ -105,22 +108,25 @@ sample({
...
+
+    const activeTokens: AssetByChains[] = [];
 
-    return tokens.reduce<AssetByChains[]>((acc, token) => {
+    for (const token of tokens) {
       const filteredChains = token.chains.filter((c) => {
         if (!connections[c.chainId]) return false;
-        const isDisabled = networkUtils.isDisabledConnection(connections[c.chainId]);
-        const hasMultiPallet = networkUtils.isMultisigSupported(chains[c.chainId].options);
+        if (networkUtils.isDisabledConnection(connections[c.chainId])) return false;
```
Comment (johnthecat) [inline src/renderer/features/assets/AssetsPortfolioView/model/portfolio-model.ts]: Same

File: src/renderer/features/assets/AssetsPortfolioView/model/portfolio-model.ts [hunk 4]
```diff
@@ -105,22 +108,25 @@ sample({
...
+        if (!isMultisigWallet) return true;
 
-        return !isDisabled && (!isMultisigWallet || hasMultiPallet);
+        return networkUtils.isMultisigSupported(chains[c.chainId].options) || multisigChainToInclude === c.chainId;
```
Comment (johnthecat) [inline src/renderer/features/assets/AssetsPortfolioView/model/portfolio-model.ts:125]: same

**Review by johnthecat — APPROVED** [review]

File: src/renderer/features/assets/AssetsPortfolioView/model/portfolio-model.ts (hunk 2 above)
Comment (tuul-wq) [reply src/renderer/features/assets/AssetsPortfolioView/model/portfolio-model.ts]: image
It's already `MsigAccount`, or you mean check index 0?

File: src/renderer/features/assets/AssetsChainView/ui/AssetsChainView.tsx (hunk 1 above)
Comment (tuul-wq) [reply src/renderer/features/assets/AssetsChainView/ui/AssetsChainView.tsx]: `if (!connections[c.chainId]) return false;`
Or what do you mean?

Comment (johnthecat) [reply src/renderer/features/assets/AssetsChainView/ui/AssetsChainView.tsx]:
```
const connection = connections[c.chainId];
if (nullable(connection)) return false
```

File: src/renderer/features/assets/AssetsPortfolioView/model/portfolio-model.ts (hunk 2 above)
Comment (johnthecat) [reply src/renderer/features/assets/AssetsPortfolioView/model/portfolio-model.ts]: index

**Pull request merged**